src/features/deposit/components/DepositUIMachineFormSyncProvider.tsx

```diff
@@ -38,17 +40,17 @@ export function DepositUIMachineFormSyncProvider({
   }, [watch, actorRef])
 
   useEffect(() => {
-    if (!userAddress) {
+    if (!userAddress || userChainType == null) {
       actorRef.send({
         type: "LOGOUT",
       })
     } else {
       actorRef.send({
         type: "LOGIN",
-        params: { userAddress, rpcUrl },
+        params: { userAddress, userChainType, rpcUrl },
```

Comment:
Just note `rpcUrl` no longer exists in the main branch. I think it's worth updating the branch to ensure that the latest PR to the main is smoothly merged.

Reply:
Be careful when pulling from the `main` branch at [src/features/machines/depositUIMachine.ts](https://github.com/defuse-protocol/defuse-sdk/pull/144/files#diff-c08af5996b118d9a146a286a055fce826f7525c4dcdf4af447e8dff5851b31b8). There might be a small conflict.